docroot/modules/custom/foia_api/src/Plugin/rest/resource/WebformSubmissionResource.php

```diff
@@ -136,6 +136,26 @@ public function post($data) {
       $this->logSubmission($statusCode, $message);
       return new ModifiedResourceResponse(['errors' => $message], $statusCode);
     }
+    // Validate emal, phone and address,
+    // any one of those three not blank, it should pass.
+    $email = $addressLine1 = $addressCity = $addressStateProvince = $addressZipPostalCode = $country = $mailingAddress = '';
+    $email = isset($data['email']) ? ($data['email'] ?? '') : $email;
+    $phoneNumber = isset($data['phone_number']) ? ($data['phone_number'] ?? '') : '';
+    if (isset($data['address_line1']) && isset($data['address_city']) && isset($data['address_state_province']) && isset($data['address_zip_postal_code']) && isset($data['address_country'])) {
+      $addressLine1 = $data['address_line1'] ?? '';
+      $addressCity = $data['address_city'] ?? '';
+      $addressStateProvince = $data['address_state_province'] ?? '';
+      $addressZipPostalCode = $data['address_zip_postal_code'] ?? '';
+      $country = $data['address_country'] ?? '';
+      $mailingAddress = (!$addressLine1 || !$addressCity || !$addressStateProvince || !$addressZipPostalCode || !$country) ? '' : 'mailling address';
+    }
+
+    if (!$mailingAddress && !$email && !$phoneNumber) {
+      $statusCode = 422;
+      $message = t("At least Email, Mailing address, or Phone are required.");
+      $this->logSubmission($statusCode, $message);
+      return new ModifiedResourceResponse(['errors' => $message], $statusCode);
```

Comment:
I believe that the 'errors' key should be pointing to an object of field names to error messages - not just a string. You can see an example returned if you submit a request via API with a missing required field. I think in our case we want it to be the same message all three fields -- something like:
```
$message = 'In order to submit your request, you must provide at least one of the following: email address, mailing address, or phone number.';
$errors = [
  'email' => $message,
  'phone' => $message,
  'address_line1' => $message,
];
```